src/function/arithmetic/multiply.js

```diff
@@ -110,11 +110,11 @@ export const createMultiply = /* #__PURE__ */ factory(name, dependencies, ({ typ
     // a dense
     const adata = a._data
     const asize = a._size
-    const adt = a._datatype
+    const adt = a._datatype === undefined ? a.getDataType() : a._datatype
```

Comment:
I think all these cases can be simplified to `a._datatype || a.getDataType()`

Reply:
I simplified them except in the case of sparse matrices, where I used 
```js
a._datatype || a._data === undefined ? a._datatype : a.getDataType()
```